### PR TITLE
Prevent mixing model parallel and multiprocessing.

### PR DESCRIPTION
**Pull request by stephenroller**
**Patch description**
Follow up to #2962. If the user attempts to combine model parallelism and multiprocessing, throw an error and point them to that issue.

**Testing steps**
Manual testing. New CI.

### CODE REVIEW

File: parlai/utils/torch.py [hunk 1]
```diff
@@ -333,6 +333,24 @@ def __init__(self):
...
+    def check_compatibility(self, opt):
+        """
+        Check compatibility for opts.
+
+        Really just used to raise an error message if the user mixes multiprocessing and
+        model parallelism.
+        """
+        if opt.get('multiprocessing') and not os.environ.get('PARLAI_FORCE_MP'):
+            raise RuntimeError(
+                "It looks like you are trying to mix multiprocessing data "
```
Comment (dianaglzrico) [inline parlai/utils/torch.py:345]: Nice message 👍

File: tests/test_distributed.py [hunk 2]
```diff
@@ -169,6 +169,23 @@ def test_chunked_teacher(self):
...
+    def test_no_model_parallel(self):
+        """
+        Checks that we throw an error when combining mp_train with.
+
+        --model-parallel true.
+        """
+        config = copy.deepcopy(self._base_config)
+        config['model_parallel'] = True
+        for m in [
+            'transformer/generator',
```
Comment (dianaglzrico) [inline tests/test_distributed.py:181]: Why these models only?

**Comment by stephenroller** [comment]
> This looks great! once I read the issue it all made sense, good idea to add this check. Left a few comments only :)

These are the only models that support it.

File: tests/test_distributed.py (hunk 2 above)
Comment (stephenroller) [reply tests/test_distributed.py:181]: These are the only models that support it, and it covers TRA, TCA, and TGA, so all the abstract classes are covered.

**Pull request merged**